microsoft/testsuites/dpdk/dpdkutil.py

```diff
@@ -610,3 +619,455 @@ def _parallel_cleanup(node: Node) -> None:
+        queue_count = 8
+    else:
+        queue_count = 16
+    # MANA supports 32 queues, however we map core/queues 1:1
+    # this is due to a bug in the logic that parses rxq/txq count
+    # MANA will reject configurations where there are more cores
+    # than queues, however we are using 2 ports so have 2x queues
+    # to assign to cores. So, we assign 2 queues per core.
+    # pro: use more queues with less cores
+    # con: 1 core is servicing queue N for both ports.
+    # TODO: change this when the txq/rxq bug is fixed.
+    # You still get much higher throughput than cx5 is capable of.
+    if is_mana:
+        queue_count *= 2
+    return queue_count
+
+
+def verify_dpdk_l3fwd_ntttcp_tcp(
+    environment: Environment,
+    log: Logger,
+    variables: Dict[str, Any],
+    pmd: str = "netvsc",
+    enable_gibibyte_hugepages: bool = False,
+    force_single_queue: bool = False,
+) -> None:
+    # This is currently the most complicated DPDK test. There is a lot that can
+    # go wrong, so we restrict the test to netvsc and only a few distros.
+    # Current usage is checking for regressions in the host, not validating all distros.
+    #
+    #  l3 forward is also _not_ intuitive, and Azure net routing doesn't help.
+    #  Azure primarily considers IP and not ethernet addresses.
+    #  It's weirdly normal to see ARP requests providing inaccurate MAC addresses.
+    #  Why? Beyond me at this time. -mm
+    #
+    # SUMMARY:
+    #  The test attempts to change this initial default LISA setup:
+    #   snd_VM      fwd_VM      rcv_VM
+    #   |s_nic1 <-> |f_nic1 <-> |r_nic1    10.0.0.0/24 (Subnet A)
+    #   |snd_nic_a <-> |f_nic2 <-> |r_nic2    10.0.1.0/24 (Subnet B)
+    #   |s_nic3 <-> |f_nic3 <-> |rcv_nic_b    10.0.2.0/24 (Subnet C)
+    # To this:
+    #    snd_VM      fwd_VM      rcv_VM
+    #    |s_nic1 <-> |f_nic1 <-> |r_nic1    10.0.0.0/24 (Subnet A)
+    #    |snd_nic_a <-> |f_nic2     |          10.0.1.0/24 (Subnet B/C)
+    #                | ↕ DPDK as NVA forwarding (and filtering) traffic
+    #                |f_nic3 <-> |rcv_nic_b   10.0.2.0/24 (Subnet B/C)
+    #
+    #  With the goal of guaranteeing that snv_VM cannot reach
+    #  rcv_VM on Subnet B/C without traffic being forwarded through
+    #  DPDK on  fwd_VM
+    #
+    # Our key objectives are:
+    # 1. intercepting traffic from  snd_nic_a bound for rcv_nic_b,
+    #      sending it to f_nic2 (az routing table)
+    #
+    # 2. forwarding from f_nic2 to f_nic3 to bridge subnet b/c (DPDK)
+    #
+    # 3. enjoy the thrill of victory, ship a cloud net applicance.
+
+    l3fwd_app_name = "dpdk-l3fwd"
+    send_side = 1
+    receive_side = 2
+    # arbitrarily pick fwd/snd/recv nodes.
+    forwarder, sender, receiver = environment.nodes.list()
+    if (
+        not isinstance(forwarder.os, Ubuntu)
+        or forwarder.os.information.version < "22.4.0"
+    ):
+        raise SkippedException("l3fwd test not compatible, use Ubuntu >= 22.04")
```

Comment:
skip the case in the beginning to save time

Reply:
This is the beginning